### PR TITLE
Added well architected md. Notes are draft for discussion.

### PR DESCRIPTION
**Pull request by jrrudge**

### CODE REVIEW

**Comment by mcnamarabrian** [comment]
Better to isolate the WAR to the application rather than the build process?

File: WELL-ARCHITECTED.md [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+* [x] **[Required]** Use temporary credentials between resources and components
+
+* [ ] None of these
+
+
+##### Notes
+
+> * We use IAM policy to ensure that resources can only be called by other resources that should be calling them.
+>
+> * The pipeline will assume a role with only the permissions it requires to deploy the application. This will either be only being able to perform a specific action on multiple resources or any action on a particular resource.
```
Comment (mcnamarabrian) [inline WELL-ARCHITECTED.md]: I think it better we call out the application in general, not the pipeline.

File: WELL-ARCHITECTED.md [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+
+---
+
+#### REL 2. How do you build resiliency into your Serverless application?
+
+* [ ] Question does not apply to this workload
+
+
+* [x] **[Required]** Manage transaction, partial, and intermittent failures
+* [ ] **[Required]** Manage duplicate and unwanted events
```
Comment (mcnamarabrian) [inline WELL-ARCHITECTED.md]: I think we have this requirement met.  We handle duplicates well - the Dynamo key and converted S3 object for each analysis is tied to the input object being analyzed.  Pushing the same document will result in the same artifact.  There could be a question regarding document names (what happens when a new document with the same name) but I think we check this box.

**Review by mcnamarabrian — CHANGES_REQUESTED** [review]
I think we should consider the WAR for the application, not the pipeline.  Also, I think we check the box for dealing with duplicate events.

File: WELL-ARCHITECTED.md (hunk 1 above)
Comment (jrrudge) [reply WELL-ARCHITECTED.md]: I updated the wording to speak in application wide terms.

File: WELL-ARCHITECTED.md (hunk 2 above)
Comment (jrrudge) [reply WELL-ARCHITECTED.md]: Checked and updated supporting text.

File: WELL-ARCHITECTED.md [hunk 3]
```diff
@@ -81,7 +81,7 @@
 
 > * We use IAM policy to ensure that resources can only be called by other resources that should be calling them.
 >
-> * The pipeline will assume a role with only the permissions it requires to deploy the application. This will either be only being able to perform a specific action on multiple resources or any action on a particular resource. 
+> * All application components will assume a role with only the permissions it requires in order to perform it's function. This will either be only being able to perform a specific action on multiple resources or any action on a particular resource.
```
Comment (mcnamarabrian) [inline WELL-ARCHITECTED.md]: (sp) `perform its function`.

**Review by mcnamarabrian — COMMENTED** [review]
Minor spelling error but otherwise :shipit:.

**Review by mcnamarabrian — APPROVED** [review]
Thanks for making the adjustment.  Once the spelling is updated feel free to :shipit:.

**Pull request merged**